Core/GDCore/IDE/Events/ExpressionCompletionFinder.h

```diff
@@ -131,28 +162,74 @@ struct ExpressionCompletionDescription {
    */
   bool IsExact() const { return isExact; }
 
+  /**
+   * \brief Return the first character index of the autocompleted part.
+   */
+  size_t GetReplacementStartPosition() const { return replacementStartPosition; }
+
+  /**
+   * \brief Return the first character index after the autocompleted part.
+   */
+  size_t GetReplacementEndPosition() const { return replacementEndPosition; }
+
+  /**
+   * \brief Set if the expession is the last child of a function call.
+   */
+  ExpressionCompletionDescription& SetIsLastParameter(bool isLastParameter_) {
+    isLastParameter = isLastParameter_;
+    return *this;
+  }
+
+  /**
+   * \brief Check if the expession is the last child of a function call.
+   */
+  bool IsLastParameter() const { return isLastParameter; }
+
+  /**
+   * \brief Set the parameter metadata.
+   */
+  ExpressionCompletionDescription& SetParameterMetadata(const gd::ParameterMetadata &parameterMetadata_) {
+    parameterMetadata = &parameterMetadata_;
+    return *this;
+  }
+
+  /**
+   * \brief Return the parameter metadata if the completion is about a parameter.
+   */
+  const gd::ParameterMetadata& GetParameterMetadata() const { return *parameterMetadata; }
```

Comment:
It's generally a good idea to be safe to use the ["null object pattern"](https://en.wikipedia.org/wiki/Null_object_pattern). 
We keep a pointer internally (so that we can modify what is pointed easily) but instead of setting it to `nullptr` when it's empty, we point it to an empty object (here, a ParameterMetadata constructed without anything and stored as a static member of the class).
This is almost equally performant (to check if the parameter metadata is defined, you just check if the pointer is not pointing to this empty object, called `badParameterMetadata`).
And this is a bit safer, in the sense that if someone tries to read the parameter metadata in a context where it's not valid, it's not crashing.

(An better alternative, safe and always correct would be to use an `Optional` like type, so that you would be forced to check if the parameter metadata is defined before using it. But we use this "null object pattern" a lot and it works well enough!)